File: content/problems/2366-minimum-replacements-to-sort-the-array/solutions/right-to-left-balanced-splitting/solution.cpp
class Solution {
    public:long long minimumReplacement(vector<int>& nums){
        long long answer=0,limit=nums.back();
        for(int i=nums.size()-2;i>=0;--i){
            long long parts=(nums[i]+limit-1)/limit;
            answer+=parts-1;
            limit=nums[i]/parts;
        }
        return answer;
    }
};

File: content/problems/2366-minimum-replacements-to-sort-the-array/solutions/right-to-left-balanced-splitting/Solution.java
class Solution {
    public long minimumReplacement(int[] nums){
        long answer=0;
        int limit=nums[nums.length-1];
        for(int i=nums.length-2;i>=0;i--){
            long parts=(nums[i]+(long)limit-1)/limit;
            answer+=parts-1;
            limit=(int)(nums[i]/parts);
        }
        return answer;
    }
}
